test/users/post.spec.js

```diff
@@ -0,0 +1,71 @@
+const request = require('supertest');
+const _ = require('lodash');
+
+const app = require('../../app');
+const { createUser, buildUserJson } = require('../factory/users_factory');
+const { FIELD_VALIDATION_ERROR, userEmailRepeatedError } = require('../../app/errors');
+const { underscoreKeys, camelizeKeys } = require('../../app/helpers/object_utils');
+const { User } = require('../../app/models');
+
+describe('POST /users', () => {
+  const httpRequest = params =>
+    request(app)
+      .post('/users')
+      .send(params);
+
+  describe('when params are OK', () => {
+    const userParams = buildUserJson();
+
+    test('Responds with 201 status code', () =>
+      userParams.then(httpRequest).then(response => expect(response.statusCode).toBe(201)));
```

Comment:
```
let userCreatedResponse = {};
beforeAll(async () => {
  userCreatedResponse = buildUserJson().then(httpRequest);
}
```
That way, the test remains only with an expect. It's cleaner, and we do the request only once.

Reply:
Idem for the rest tests

Reply:
I forgot to mention it, but the card asks for password restrictions validation. It's useful to do the checks, so you know when we are forgetting something